operations/dataGenV2.go

```diff
@@ -0,0 +1,332 @@
+		UserAttributes map[string]string `json:"user_properties"`
+		EventAttributes map[string]string `json:"event_properties"`
+	}
+
+	var op output
+	op.UserId = userId
+	op.Event = event
+	op.Timestamp, _ = strconv.Atoi(fmt.Sprintf("%v", segmentConfig.Start_Time.Add(time.Second * time.Duration(eventCounter * segmentConfig.Activity_ticker_in_seconds)).Unix()))
+	op.UserAttributes = userAttributes
+	op.EventAttributes = eventAttributes
+	e, _ := json.Marshal(&op)
+	return string(e)
+}
+
+func WaitIfRealTime(duration int) {
+	if(IsRealTime() == true){
+		time.Sleep(time.Duration(duration) * time.Second)
+	}
+}
+
+func IsRealTime() bool {
+	if(config.ConfigV2.Real_Time == true){
+		return true
+	}
+	return false
+}
+func SetUserAttributes(segmentConfig config.UserSegmentV2, userId string) map[string]string{
+	var userAttributes map[string]string
+	if(segmentConfig.Set_attributes == true){
+		attr := segmentConfig.User_attributes[userId]
+		if(attr != nil){
+			userAttributes = attr
+		}
+	}
+	return userAttributes
+}
+
+func SetEventAttributes(segmentConfig config.UserSegmentV2,eventName string) map[string]string{
+	var eventAttributes map[string]string
+	if(segmentConfig.Set_attributes == true){
+		attr := segmentConfig.Event_attributes[eventName]
+		if(attr != nil){
+			eventAttributes = attr
+		}
+	}
+	return eventAttributes
+}
+
+func GetRandomActivity(probMap SegmentProbMap) string {
+	activity := rand.Intn(probMap.activityMultiplier)
+	activityName, state := probMap.activityProbRangeMap.Get(activity)
+	if(state == false){
+		Log.Error.Fatal(fmt.Sprintf("ActivityProbablityRangeMap: Key not found %v", activity))
+	}
+	return activityName
+}
+
+func GetRandomEvent(probMap SegmentProbMap) string {
+	event := rand.Intn(probMap.eventMultiplier)
+	eventName, state := probMap.eventProbRangeMap.Get(event)
+	if(state == false){
+		Log.Error.Fatal(fmt.Sprintf("EventProbablityRangeMap: Key not found %v", event))
+	}
+	return eventName
+}
+
+func GetRandomEventWithCorrelation(lastKnownGoodState *string, seedEvents []string, probMap SegmentProbMap) (string) {
+	if *lastKnownGoodState == "" {
+		*lastKnownGoodState = seedEvents[rand.Intn(len(seedEvents))]
+		return *lastKnownGoodState
+	}
+    
+	event := rand.Intn(probMap.EventCorrelationMultiplier[*lastKnownGoodState])
+	eventName, state := probMap.EventCorrelationRangeMap[*lastKnownGoodState].Get(event)
+	if(state == false){
+		Log.Error.Fatal(fmt.Sprintf("EventProbablityRangeMapWithCorrelation: Key not found %v", event))
+	}
+	*lastKnownGoodState = eventName
+	return eventName
+}
+
+func ComputeRangeMap(probMap map[string]float64) (utils.RangeMap, int) {
+
+	// Assuming sum of the probablities of elements is 1
+	// TODO janani: yet to calculate relative probablities if the sum is not 1
+	min := 1.0
+		
+	//TODO call this from util once you find a way to iterate values
+	for _, element := range probMap {
+		if element < min && element != 0{
+			min = element
+		}
+	}
+
+	multiplier := 1.0
+	temp := 0.0
```

Comment:
multiplier = ceil(1 / min) ?
Also need to handle the case if min is 0.

Reply:
Initialization is done with min = 1.0